portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -39,29 +42,37 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     int maxComments = Integer.parseInt(request.getParameter("maxComments"));
+    // TODO: sort by timestamp
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder.withLimit(maxComments));
     List<String> comments = new ArrayList<>();
     for (Entity entity : results) {
-      comments.add((String) entity.getProperty("comment"));
+        String text = (String) entity.getProperty("text");
+        comments.add(text);
     }
-    response.setContentType("application/json;");
+    response.setContentType("application/json");
     response.getWriter().println(new Gson().toJson(comments));
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the input from the form
-    String comment = request.getParameter("comment");
+    String text = request.getParameter("comment");
+    String maxComments = request.getParameter("numOfCom");
     long timestamp = System.currentTimeMillis();
 
     //Create entity for Datastore
     Entity commentEntity = new Entity("Comment");
-    commentEntity.setProperty("comment", comment);
+    commentEntity.setProperty("text", text);
     commentEntity.setProperty("timestamp", timestamp);
     datastore.put(commentEntity);
 
+    //If no maxComments was selected pass 1 as default
+    if (!maxComments.equals("5") && !maxComments.equals("10")) {
```

Comment:
What if maxComments is `null`? 
Also, if someone provided the value 7, is that a problem? I think that it should be left at "7" (think, for example, you want to change the UI but not change the server implementation).

A more flexible approach is to allow any *number* to be provided in the original request.

Reply:
You are right, it was some kind of bandage fix so previous number of comments will remain selected after a comment submition.. I changed it to something that still works that I think is much better

Reply:
Looks good.
Do note that having `null` as an input still causes the server to throw an exception in many cases.

Reply:
Since I added the "selected = selected" for the "1" option in the maxcomments select the input should never be null because if nothing is selected it's 1. Anyway I added something that can catch this null issue on dataServlet just in case